hot/loader-helpers.js

```diff
@@ -0,0 +1,25 @@
+/* eslint-env commonjs */
+const path = require(`path`);
+
+// Retrieve elemName for hot injection from path convention
+//
+// elem name patterns look like this
+//    ./src/.../${elemName}/index.jade
+// OR ./src/.../${elemName}.jade
```

Comment:
Another off the cuff thought for a future revision... Our naming convention works well when the different component subparts (template, style, 'controller') are distinguished by file type/extension, but it breaks down otherwise. For instance, back in the earliest days we used Jade for most templates but wrote some of the simpler ones directly with hyperscript in JS (`h('div')` etc.). A scheme that's resilient to using any file type/tooling could include info in the filename too, so that rather than `index.jade` it'd be `template.jade` or `template.js` or `template.hbs` or whatever. So a component dir might look like:
```
my-widget/
  index.js  // imports component class and registers custom element
  component.js  // defines and exports the component class but doesn't register it
  [controller.js]
  styles.*
  template.*
```